test/threadpools.jl

```diff
@@ -0,0 +1,44 @@
+# This file is a part of Julia. License is MIT: https://julialang.org/license
+
+using Test
+using Base.Threads
+
+#=
+Test to ensure that the interactive threadpool works as designed. Spawn two
+tasks A and two tasks B. A is a standard task that does a lot of work (~2
+seconds) without yielding. B is an interactive task that does a little work
+(~0.02 seconds) and yields, 100 times. The two A tasks should consume two
+threads (for 2 seconds) without allowing either B task to run. In the B task,
+on each of the 100 iterations, we test to ensure that the time taken to do
+the little work (0.02 seconds) as well as yielding (i.e. switching to the
+other B task, doing the little work there and switching back) does not exceed
+0.15 seconds. This should (and does) fail without any threads assigned to the
+interactive threadpool.
+=#
+const N = 263000000 # busywork(N) takes ~1 sec on an i7-9750H @ 2.6GHz
+function busywork(n::Int)
+    acc = 0
+    for i = 1:n
+        x = rand(2:10)
+        acc += i * x
+    end
+    return acc
+end
+
+function itask()
+    h = N ÷ 50
+    for i = 1:100
+        t1 = time()
+        busywork(h)
+        yield()
+        t2 = time()
+        @test t2 - t1 < 0.15
```

Comment:
This sort of test can have trouble in CI but I'm not sure what else to do.

Reply:
I've disabled this test from running (but left the test file in there) and added a couple of other tests (is a task spawned into the correct pool), etc.